api/src/test/java/org/openmrs/test/BaseContextSensitiveTest.java

```diff
@@ -353,10 +354,17 @@ public Properties getRuntimeProperties() {
 		runtimeProperties.setProperty(ModuleConstants.IGNORE_CORE_MODULES_PROPERTY, "true");
 		
 		try {
-			File tempappdir = File.createTempFile("appdir-for-unit-tests-", "");
+			final File tempappdir = File.createTempFile("appdir-for-unit-tests-", "");
```

Comment:
did you try to use the approach I outlined at https://talk.openmrs.org/t/unit-test-files-left-in-place/29617/2?u=teleivo ?

The way we create the temp dir is just not the JUnit 4 and 5 way. I would prefer we try to stick with what is the preferred way of the testing libraries. Its less code and easier to read and understand.

If it does not work and/or we have legitimate reasons to do this our own way I am ok with it.

Reply:
I actually think that using a JUnit 4 rule might require more steps to make this work. The problem is that we seem to only invoke this `getRuntimeProperties()` method during class construction and it then assumes that the `APPLICATION_DATA_DIRECTORY_RUNTIME_PROPERTY` stays stable across tests in the same class. I'm not sure the TempFolder rule allows for that?

With the JUnit 5 tests, I think we could use `@TempDir` correctly, since that can be scoped to "all tests in the same class."

Reply:
there is a `@ClassRule`
https://junit.org/junit4/javadoc/4.12/org/junit/ClassRule.html

that can probably help us achieve that.

And should it fail if it cannot delete the folder? Because by default it does not unless we use what is explained here

https://www.baeldung.com/junit-4-rules#1-the-temporaryfolder-rule

Reply:
so in JUnit 4 it seems that to achieve the same behavior as in JUnit 5 TempDir

we would have to do something like

```java
@ClassRule
public static TemporaryFolder folder = TemporaryFolder.builder().assureDeletion().build();
```